python/dgl/mock_sparse2/reduction.py

```diff
@@ -0,0 +1,322 @@
+"""DGL sparse matrix reduce operators."""
+from typing import Optional
+
+import torch
+
+from .sparse_matrix import SparseMatrix
+
+
+def reduce(A: SparseMatrix, dim: Optional[int] = None, rtype: str = "sum"):
```

Comment:
Is it necessary to provide `reduce` other than the specific ops?

Reply:
It's in `mock_sparse` so I added it.